pkg/certificate/types.go

```diff
@@ -82,13 +84,13 @@ type Certificate struct {
 	signingIssuerID    string
 	validatingIssuerID string
 
-	certType CertType
+	certType certType
 }
 
 // Issuer is the interface for a certificate authority that can issue certificates from a given root certificate.
 type Issuer interface {
 	// IssueCertificate issues a new certificate.
-	IssueCertificate(CommonName, time.Duration) (*Certificate, error)
+	IssueCertificate(IssueOptions) (*Certificate, error)
```

Comment:
nit: might have this take an interface with the single method GetCommonName()?

Makes it a bit easier to look at and realize that IssueOptions has no other exported fields/methods

Reply:
I considered this. There is one exported field `ValidityDuration` which would need to go to `getValidityDuration` and will eventually add `GetSpiffeId` to the interface.  Not sure the best approach. So open to suggestions

Reply:
I think it was in reference to `IssueOptions`. After thinking some more, I believe I choose a struct since it is really just data not an implementation of a set of functionality but I realize that is kind of fuzzy.  Happy to adjust if you anyone has strong opinion here.

Reply:
Ya that sounds reasonable to me!